Source/ASDisplayNode.mm

```diff
@@ -584,14 +590,21 @@ - (BOOL)isNodeLoaded
   if (ASDisplayNodeThreadIsMain()) {
     // Because the view and layer can only be created and destroyed on Main, that is also the only thread
     // where the state of this property can change. As an optimization, we can avoid locking.
-    return [self _locked_isNodeLoaded];
+    return [self _unsafe_unlocked_isNodeLoaded];
   } else {
     ASDN::MutexLocker l(__instanceLock__);
     return [self _locked_isNodeLoaded];
   }
 }
 
 - (BOOL)_locked_isNodeLoaded
+{
+  ASAssertLocked(__instanceLock__);
+  return (_view != nil || (_layer != nil && _flags.layerBacked));
+}
+
+/// Called without the lock. Client is responsible for locking safety.
+- (BOOL)_unsafe_unlocked_isNodeLoaded
```

Comment:
Can we simplify all of this by just using `_layer != nil` inline when we want a fast loaded check?

Reply:
great point!